[Merl 491] - Document `getNextServerSideProps` and `getNextStaticProps` (#1076)
wpengine/faustjs: internal/faustjs.org/docs/next/reference/getNextStaticProps.mdx

---
slug: /next/reference/getNextStaticProps
title: getNextStaticProps
description: The getNextStaticProps function lets you build a static site with your WordPress data outside of the template hierarchy..
---

The `getNextStaticProps` function lets you build a static site with your WordPress data outside of the template hierarchy using Next.js file based pages. The function should be returned from `getStaticProps`:

```tsx
export async function getStaticProps(context: GetStaticPropsContext) {
  return getNextStaticProps(context, {
    Page: MyPage,
  });
}
```

The `getNextStaticProps` function accepts two arguments: the server side props `context`, and an object (type `GetNextStaticPropsConfig` which extends the `GetNextServerSidePropsConfig` type) with a `Page` key.
This should be your Next.js page component or any object that contains the following properties: 'query' and `variables`.

> The reason `MyPage` are passed to `getNextStaticProps` is because
> under the hood Faust will use the Apollo Client to fetch the data using the 'query' and 'variables'
> properties. If the `Page` does not have those properties then no queries will be performed.

Reviewer: ```suggestion
Faust uses `MyPage` passed via `getNextStaticProps` to fetch the data using the 'query' and 'variables' properties. If the `Page` does not have those properties then no queries will be performed.
```

> This is merely a convenient way to fetch the page data while reusing the logic between components. If you are interested on how this is done you can review the [code section here](https://github.com/wpengine/faustjs/blob/canary/packages/faustwp-core/src/getProps.ts#L35)

Reviewer: ```suggestion
This is merely a convenient way to fetch the page data while reusing the logic between components. If you are interested on how this is done you can review the [source code here.](https://github.com/wpengine/faustjs/blob/canary/packages/faustwp-core/src/getProps.ts#L35)
```
